pyproject.toml

```diff
@@ -1,3 +1,77 @@
 [build-system]
-requires = ["setuptools >= 40.8.0", "wheel"]
+requires = ["setuptools>=61.0.0", "wheel"]
 build-backend = "setuptools.build_meta"
+
+[project]
+name = "{{ project_name }}"
+version = "0.1.0"
+description = "TODO"
+authors = [
+    {name = "Maykin Media", email = "support@maykinmedia.nl"}
+]
+readme = "README.rst"
+license = {file = "LICENSE"}
+keywords = ["TODO"]
+classifiers = [
+    "Development Status :: 3 - Alpha",
+    "Framework :: Django",
+    "Framework :: Django :: 3.2",
+    "Intended Audience :: Developers",
+    "Operating System :: Unix",
+    "Operating System :: MacOS",
+    "Operating System :: Microsoft :: Windows",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Topic :: Software Development :: Libraries :: Python Modules",
+]
+requires-python = ">=3.8"
+dependencies = [
+    "django>=3.2"
+]
+
+[project.urls]
+Homepage = "https://github.com/maykinmedia/{{ project_name|lower }}"
+Documentation = "http://{{ project_name|lower }}.readthedocs.io/en/latest/"
+"Bug Tracker" = "https://github.com/maykinmedia/{{ project_name|lower }}/issues"
+"Source Code" = "https://github.com/maykinmedia/{{ project_name|lower }}"
+
+[project.optional-dependencies]
+test = [
+    "pytest",
+    "pytest-django",
+    "tox",
+    "isort",
+    "black",
+    "flake8",
+]
+coverage = [
+    "pytest-cov",
+]
+docs = [
+    "sphinx",
+    "sphinx-rtd-theme",
+]
+release = [
+    "bump-my-version",
+    "twine",
+]
+
+[tool.isort]
+profile = "black"
+combine_as_imports = true
+known_django = "django"
+known_first_party="{{ project_name|lower }}"
+sections=["FUTURE", "STDLIB", "DJANGO", "THIRDPARTY", "FIRSTPARTY", "LOCALFOLDER"]
+
+[tool.pytest.ini_options]
+testpaths = ["tests"]
+DJANGO_SETTINGS_MODULE = "testapp.settings"
+
+[tool.bumpversion]
+current_version = "0.1.0"
+files = [
+    {filename = "pyproject.toml"},
+    {filename = "README.rst"},
+    {filename = "docs/conf.py"},
+]
```

Comment:
the reason it was in a separate file is that bumpversion rewrites the `setup.cfg` file and always messes up the formatting. Is that resolved when using toml? Otherwise the config belongs in a separate file.

I'm also increasingly annoyed by bumpversion when trying to do pre-releases and have started using tbump instead.

Reply:
This is the configuration for [callowayproject/bump-my-version](https://github.com/callowayproject/bump-my-version), which is a fork of a fork of the currently used `bumpversion`. Can you describe what was the issue with pre-releases so that I can test with this new tool?

Reply:
in Open Forms (currently in Alpha), try running `bumpversion pre`

You'll see that it changes the version from `2.5.0-alpha.0` to `2.5.0-beta.0-alpha.0`. The serializing goes wrong.

Additionally, dropping the pre-release part again doesn't seem to be intuitive/possible either - try running `bumpversion minor`, and you'll see it retains the pre-release part and bumps to 2.6.0-alpha.0 instead of to `2.5.0`.